src/Pack/Core/Types.idr

```diff
@@ -407,6 +407,74 @@ export
 dependencies : Desc t -> List PkgName
 dependencies d = dependencies d.desc
 
+--------------------------------------------------------------------------------
+--          CmdArg
+--------------------------------------------------------------------------------
+
```

Comment:
Please add some docstrings to these new types and functions: What do they represent, why are they necessary, how should the be used.

Reply:
Done